kwikapi/client.py

```diff
@@ -110,33 +114,36 @@ def _make_request(self, url, post_body, headers):
         if self._stream:
             proto = PROTOCOLS[self._protocol]
             res = proto.deserialize_stream(res)
-            res = Client._extract_stream_response(res)
+            res = Client._extract_stream_response(res, self._raise_exception)
         else:
-            res = self._deserialize_response(res.read(), self._protocol)
+            res = self._deserialize_response(res.read(), self._protocol,
+                    self._raise_exception)
 
         return res
 
     @staticmethod
-    def _deserialize_response(data, protocol):
+    def _deserialize_response(data, protocol, raise_exception=True):
         proto = PROTOCOLS[protocol]
         r = proto.deserialize(data)
-        return Client._extract_response(r)
+        return Client._extract_response(r, raise_exception)
 
     @staticmethod
-    def _extract_response(r):
+    def _extract_response(r, raise_exception=True):
         success = r['success']
         if not success:
             r.pop('success')
-            raise Exception(r) # FIXME: raise proper exc
+            r = ResponseError(r)
```

Comment:
@prashanthellina 
Can you suggest any better naming for `ResponseError`, if required?

Reply:
@pavan-maddula This sounds alright. Thank you.